draft-ietf-scitt-architecture.md

```diff
@@ -371,6 +371,156 @@ Issuers SHOULD use consistent decentralized identifiers for all their Statements
 They MAY update their DID manifest, for instance to refresh their signing keys or algorithms, but they SHOULD NOT remove or change any prior keys unless they intend to revoke all Signed Statements that are registered as Transparent Statements issued with those keys.
 This DID appears in the Issuer protected header of Signed Statements' Envelopes, while the version of the key from the manifest used to sign the Signed Statement is written in the `kid` header.
 
+`kid` MUST either be an absolute URL,
+or `iss` MUST also be present in the protected header,
+and `kid` MUST be a URL relative to `iss`.
+
+Resolving `kid` MUST return an Identity Document of a registered content type (a set of public keys).
+In the case of `kid` being an absolute DID URL, the identity document is called a DID Document,
+and is expected ot have content type `application/did+json`.
+
+Dereferencing a DID URL requires that it first be resolved, and then the fragment processed according to the media type.
+
+For example, when resolving `did:example:123#key-42`,
+first the identity document for `did:example:123` is resolved as content type `application/did+json`,
+next, the fragment `#key-2` is dereferenced to a verification method that contains a `publicKeyJwk` property in JSON.
+
+The content type of `publicKeyJwk` is expected to be `application/jwk+json`.
+
+The details of both `did resolution` and `did dereferencing` are out of scope for this document.
+
+The `iss` or `kid`, might not be DID URLs, however the following interfaces MUST be satisfied in order to ensure
+issuer identity documents, and associated keys are discoverable in a consistent manner.
+
+#### Resolving Identity Documents
+
+The value of `id` might be found the `iss` or `sub` claims if they are present in the protected header or payload.
+
+```
+resolve = (id: string, accept: content_type = 'application/did+json') =>
+idDocument (of content type application/did+json).
+```
+
+For example:
+
+```
+did:example:123
+```
+
+Might resolve to:
+
+```
+{
+  "id": "did:example:123",
+  "verificationMethod": [{
+    "id": "#key-42",
+    "type": "JsonWebkey",
+    "controller": "did:example:123",
+    "publicKeyJwk": {
+      "kty": "EC",
+      "crv": "P-384",
+      "alg": "ES384",
+      "x": "LCeAt2sW36j94wuFP0gNEIHDzqR6Nh_Udu2ObLer3cKFBCaAHY1svmbPV69bP3RH",
+      "y": "zz2SkcOGYM6PbYlw19tcbpzo6bEMYHIwGBnN5rd8QWykAprstPdxx4U0uScvDcYd"
+    }
+  }]
+}
+```
+
+Editor note, we might wish to eliminate this intermediate identity document content type,
```

Comment:
@Sakurann this is related to the conversations we have had on https://www.w3.org/TR/did-core/#verification-relationships

... if you use those JSON-LD relationships would cannot eliminate did documents, or you need to encode the relationship in the identifier itself... for example:

```
did:example:123/assertion/kid => jwk+json
did:example:123/assertion=> jwk-set+json
```